### DISCUSSION TITLE
Supabase client in nextjs 14 returning old data

### DISCUSSION THREAD

**Discussion by TomVH** [supabase/supabase#19916] [Questions]
I've created an application that stores information about itineraries, selecting itineraries for a specific user as follows:
const { data, error } = await supabase.from('itinerariesv3').select('title').eq('userid', userId);

gives me the rows belonging to that specific user. It's currently showing old data of yesterday but my table has changed multiple times since then due to my testing.

When duplicating this table and choosing this to perform the select from, it will do it once with the newest data but afterwards it won't show changed data.

I'm currently using the server actions in nextjs and somehow data is being cached. I don't think it's on nextjs level because I've tried to restart the application, restarted my database. When calling the data for a second time I also don't see any new hits in the API Edge Network logs. When I change the columns to a specific column I will see a new log for that specific query. When performed again, no new log is generated.

Any ideas on how I can make sure that everytime I call this API, it retrieves the newest values and it does not get cached somewhere?
Again, I also tried to add the export const dynamic = "force-dynamic"; in my rootlayout but that didn't help much either.

Thanks in advance!

**Comment by GaryAustin1** [comment]
This is a next.js caching thing.  It caches URL's, which is why when you change the column it busts through.

https://github.com/orgs/supabase/discussions/19084

You tried on solution listed there.  You are best researching from next.js side on caching if no one comes along with a specific next.js solution for you here.
